chop/optim.py

```diff
@@ -424,3 +429,111 @@ def minimize_frank_wolfe(closure, x0, lmo, step='sublinear',
     fval, grad = closure(x)
     return optimize.OptimizeResult(x=x, nit=it, fval=fval, grad=grad,
                                    certificate=cert)
+
+def minimize_alternating_fw_prox(closure, x0, y0, prox=None, lmo=None, L0=1e-3,
+                                 step='sublinear', max_iter=200, callback=None,
+                                 *args, **kwargs):
+    """
+    Implements algorithm from [Garber et al. 2018]
+    https://arxiv.org/abs/1802.05581
+
+    to solve the following problem
+
+
+    ..math::
+        \min_{x, y} f(x + y) + R_x(x) + R_y(y).
+
+    We suppose that $f$ is $L$-smooth and that
+    we have access to the following operators:
+    
+      - a generalized LMO for $R_y$: 
+    ..math::
+        gLMO(w) = \text{argmin}_w R_y(w) + \langle w, \nabla f(x_t + y_t) \rangle
+
+      - a prox operator for $R_x$:
+    ..math::
+        prox(v) = \text{argmin}_v R_x(v) + \langle v, \nabla f(x_t+ y_t) \rangle + \frac{\gamma_t L}{2} \|v + w_t - (x_t + y_t)\|^2
+    
+    Args:
+      x0: torch.Tensor of shape (batch_size, *)
+        starting point for x
+
+      y0: torch.Tensor of shape (batch_size, *)
+        starting point for y
+      
+      prox: function
+        proximal operator for R_x
+
+      lmo: function
+        generalized LMO operator for R_y. If R_y is an indicator function,
+        it reduces to the usual LMO operator.
+
+      L0: float
```

Comment:
L0 could be mistaken with the L0 norm. How about ```lipschitz``` or ```init_lipschitz```  or ```lipschitz_guess``` . I called it lipschitz in http://openo.pt/copt/generated/copt.minimize_frank_wolfe.html#copt.minimize_frank_wolfe

Reply:
Updated, thanks!